src/nupic/ntypes/Array.hpp

```diff
@@ -319,4 +320,9 @@ class Array : public ArrayBase {
 };
 } // namespace nupic
 
+CEREAL_REGISTER_TYPE(nupic::Array)
+CEREAL_REGISTER_POLYMORPHIC_RELATION(nupic::ArrayBase, nupic::Array)
```

Comment:
this is some advanced stuff, I hope it's only for the Array/Base, which are complicated. 
Maybe add a comment that this is for the complicated pointer stuff, otherwise not needed

Reply:
Any time you have a base class and an instance class and if you serialize the instance and some of the fields you want serialized happen to be in the base class, then you apparently need to do this.  So far Array and ArrayBase are the only place that this seems to be the case.

Reply:
> So far Array and ArrayBase are the only place that this seems to be the case. 

Encoders & BaseEncoder too. But that's not that bad once you know about it :+1: